test/suites/sharing/ram_share_test.go

```diff
@@ -0,0 +1,433 @@
+		createSNInput := &vpclattice.CreateServiceNetworkInput{
+			Name: aws.String(randomName),
+			Tags: k8sTestTags,
+		}
+		createSNResult, err := secondaryVpcLatticeClient.CreateServiceNetworkWithContext(ctx, createSNInput)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Share service network to primary account using randomName
+		createShareInput := &ram.CreateResourceShareInput{
+			Name: aws.String(randomName),
+			ResourceArns: []*string{createSNResult.Arn},
+			Principals: []*string{aws.String(config.AccountID)},
+			AllowExternalPrincipals: aws.Bool(true),
+			Tags: k8sRamTestTags,
+		}
+		_, err = secondaryRamClient.CreateResourceShareWithContext(ctx, createShareInput)
+		Expect(err).To(BeNil())
+
+		// Wait for resource share invitation to appear in primary account
+		var invitation *ram.ResourceShareInvitation = nil
+		Eventually(func(g Gomega) {
+			listInvitationsInput := &ram.GetResourceShareInvitationsInput{}
+			listInvitationsResult, err := primaryRamClient.GetResourceShareInvitations(listInvitationsInput)
+			Expect(err).NotTo(HaveOccurred())
+
+			for _, inv := range listInvitationsResult.ResourceShareInvitations {
+				if *inv.ResourceShareName == randomName && *inv.SenderAccountId == secondaryAccountId {
+					invitation = inv
+					break
+				}
+			}
+			g.Expect(invitation).ToNot(BeNil())
+		}).Should(Succeed())
+
+		// Accept resource share invitation
+		acceptInput := &ram.AcceptResourceShareInvitationInput{
+			ResourceShareInvitationArn: invitation.ResourceShareInvitationArn,
+		}
+		_, err = primaryRamClient.AcceptResourceShareInvitation(acceptInput)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Wait for service network to appear in primary account
+		Eventually(func(g Gomega) {
+			getSNInput := &vpclattice.GetServiceNetworkInput{
+				ServiceNetworkIdentifier: createSNResult.Id,
+			}
+			_, err = primaryVpcLatticeClient.GetServiceNetworkWithContext(ctx, getSNInput)
+			g.Expect(err).ToNot(HaveOccurred())
+		}).WithTimeout(5 * time.Minute).WithPolling(5 * time.Second).Should(Succeed())
+
+		// Create gateway with same name as service network
+		gateway := &gwv1.Gateway{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      randomName,
+				Namespace: k8snamespace,
+			},
+			Spec: gwv1.GatewaySpec{
+				GatewayClassName: "amazon-vpc-lattice",
+				Listeners: []gwv1.Listener{
+					{
+						Name: gwv1.SectionName(randomName),
+						Port:     gwv1.PortNumber(80),
+						Protocol: gwv1.HTTPProtocolType,
+					},
+				},
+			},
+		}
+		err = testFramework.Create(ctx, gateway)
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func(g Gomega) {
+			// Policy status should be Accepted
+			gwNamespacedName := types.NamespacedName{
+				Name:      gateway.Name,
+				Namespace: gateway.Namespace,
+			}
+			gw := &gwv1.Gateway{}
+			err := testFramework.Client.Get(ctx, gwNamespacedName, gw)
+			g.Expect(err).To(BeNil())
+			g.Expect(len(gw.Status.Conditions)).To(BeEquivalentTo(2))
```

Comment:
g.Expect(len(gw.Status.Conditions)).To(BeEquivalentTo(2))

not correct assertion, it should be at least 2. Also not related to RAM sharing test. Can be removed.

Reply:
Good point, will do